src/ucp/proto/proto_init.c

```diff
@@ -648,60 +651,94 @@ ucp_proto_common_init_recv_perf(const ucp_proto_common_init_params_t *params,
-{
-    ucp_proto_caps_t *caps = params->super.caps;
+static ucs_status_t
+ucp_proto_init_single_frag_ranges(const ucp_proto_common_init_params_t *params,
+                                  const ucp_proto_common_tl_perf_t *tl_perf,
+                                  ucp_proto_perf_node_t *const tl_perf_node,
+                                  ucp_md_map_t reg_md_map, size_t range_start,
+                                  size_t frag_size) {
+    int hdr_only = (frag_size == 0);
     ucp_proto_perf_range_t xfer_perf, send_perf, recv_perf;
     const ucp_proto_perf_range_t *parallel_stages[3];
     ucs_status_t status;
-    size_t frag_size;
-
-    ucs_trace("caps" UCP_PROTO_TIME_FMT(send_pre_overhead)
-              UCP_PROTO_TIME_FMT(send_post_overhead)
-              UCP_PROTO_TIME_FMT(recv_overhead) UCP_PROTO_TIME_FMT(latency),
-              UCP_PROTO_TIME_ARG(tl_perf->send_pre_overhead),
-              UCP_PROTO_TIME_ARG(tl_perf->send_post_overhead),
-              UCP_PROTO_TIME_ARG(tl_perf->recv_overhead),
-              UCP_PROTO_TIME_ARG(tl_perf->latency));
 
     /* Network transfer time */
     ucp_proto_common_init_xfer_perf(params, tl_perf, tl_perf_node, &xfer_perf);
 
     /* Sender overhead */
     status = ucp_proto_common_init_send_perf(params, tl_perf, reg_md_map,
-                                             &send_perf);
+                                             hdr_only, &send_perf);
     if (status != UCS_OK) {
         goto out_deref_xfer_perf;
     }
 
     /* Receiver overhead */
     status = ucp_proto_common_init_recv_perf(params, tl_perf, reg_md_map,
-                                             &recv_perf);
+                                             hdr_only, &recv_perf);
     if (status != UCS_OK) {
         goto out_deref_send_perf;
     }
 
-    /* Get fragment size */
-    ucs_assert(tl_perf->max_frag >= params->hdr_size);
-    frag_size = ucs_min(params->max_length,
-                        tl_perf->max_frag - params->hdr_size);
+    parallel_stages[0] = &send_perf;
+    parallel_stages[1] = &xfer_perf;
+    parallel_stages[2] = &recv_perf;
+
+    /* Add ranges representing sending single fragment */
+    status = ucp_proto_init_parallel_stages(params, range_start, frag_size,
+                                            frag_size, 0.0, parallel_stages, 3);
+
+    ucp_proto_perf_node_deref(&recv_perf.node);
+out_deref_send_perf:
+    ucp_proto_perf_node_deref(&send_perf.node);
+out_deref_xfer_perf:
+    ucp_proto_perf_node_deref(&xfer_perf.node);
+    return status;
+}
+
+ucs_status_t
+ucp_proto_common_init_caps(const ucp_proto_common_init_params_t *params,
+                           const ucp_proto_common_tl_perf_t *tl_perf,
+                           ucp_proto_perf_node_t *const tl_perf_node,
+                           ucp_md_map_t reg_md_map)
+{
+    ucp_proto_caps_t *caps = params->super.caps;
+    ucs_status_t status;
+    size_t frag_size;
+    ucs_trace("caps" UCP_PROTO_TIME_FMT(send_pre_overhead)
```

Comment:
space line before

Reply:
Space line before already included to `UCP_PROTO_TIME_FMT`

Reply:
I mean before there ucs_trace

Reply:
Sorry, I misread it first time. Done.